### PR TITLE
added boolean %b to headers for MADNG output

### PR DESCRIPTION
**Pull request by fscarlier**
Added boolean %b to headers since MADNG is using it in its twiss files. Added test in reader as well. Only for headers at the moment, since there is no usecase yet for boolean columns

### CODE REVIEW

File: tests/test_reader.py [hunk 1]
```diff
@@ -137,6 +137,17 @@ def test_read_file_without_header(self, _tfs_file_without_header, _tfs_file_path
...
         assert_frame_equal(df, df_for_compare)
 
+    def test_read_file_with_bool_in_header(self, _tfs_file_with_bool_in_header, _tfs_file_pathlib):
+        df = read_tfs(_tfs_file_with_bool_in_header)
+        assert df.headers['BOOLTRUE1'] == True
+        assert df.headers['BOOLTRUE2'] == True
+        assert df.headers['BOOLTRUE3'] == True
+        assert df.headers['BOOLFALSE1'] == False
+        assert df.headers['BOOLFALSE2'] == False
+        assert df.headers['BOOLFALSE3'] == False
```
Comment (JoschD) [inline tests/test_reader.py:147]: I think these should be `is True` and `is False`.

**Comment by fscarlier** [comment]
I agree with @JoschD. Madng is slowly being used and tested, and it will probably replace PTC for normal form/rdt use cases quite soon. Small packages like tfs-pandas should be adapted to help any transition. The "madx" switch could be a good option indeed.

**Comment by fsoubelet** [comment]
Then this could be the opportunity for a few PRs before the next release. I like a switch that would default to `MAD-X` for now and make the user aware.

> Also, I'm not a fan of a warning. Looks like pythonic madx..

We can log and raise an error if we feel like this is the best option, I was merely suggesting ;)

**Comment by JoschD** [comment]
I would just adapt the "validate" function and the validate-switch, such that you can turn it on/off as before but also give it as "MAD-NG" or "MAD-X" and it checks for different things. Would be a minor change in that regard. 

The big change would be to always convert to the new datatypes `%b` for boolean and `%i` for complex.  If the user wants MAD-X compatibility they need to change the dtype of the respective columns/headers or split the value in real/imag manually.

**Pull request closed without merging**